formatting/src/main/java/com/avast/metrics/dropwizard/formatting/Formatter.java

```diff
@@ -0,0 +1,16 @@
+package com.avast.metrics.dropwizard.formatting;
+
+/**
+ * Generic formatter of metric names and values.
+ */
+public interface Formatter {
+    /**
+     * Separator of name parts.
+     */
+    String nameSeparator();
+
+    /**
+     * Sanitize part of a metric name to not contain any illegal character. Escape, replace or delete them.
+     */
+    String sanitizeName(String namePart);
```

Comment:
No, it's primary purpose is to format metrics to a string form. I agree that this may not be obvious from this revision, see links below. Method sanitizeName() is used to fix invalid/unsupported characters in the name that would break format of the destination system.

https://github.com/mixalturek/metrics/blob/7b29d844af253b279d8917a07300c1288459d728/formatting/src/main/java/com/avast/metrics/dropwizard/formatting/Formatter.java
https://github.com/mixalturek/metrics/blob/7b29d844af253b279d8917a07300c1288459d728/formatting/src/main/java/com/avast/metrics/dropwizard/formatting/GraphiteFormatter.java